sacc2/sacc.py

```diff
@@ -0,0 +1,871 @@
+        if S.covariance is not None:
+            hdus.append(S.covariance.to_hdu())
+
+        # Make and save the final FITS data
+        hdu_list = fits.HDUList(hdus)
+        hdu_list.writeto(filename, overwrite=overwrite)
+
+    @classmethod
+    def load_fits(cls, filename):
+        """
+        Load a Sacc data set from a FITS file.
+
+        Don't try to make these FITS files yourself - use the tools
+        provided in this package to make and save them.
+
+        Parameters
+        ----------
+
+        filename: str
+            A FITS format sacc file
+
+        """
+        hdu_list = fits.open(filename)
+
+        # Split the HDU's into the different sacc types
+        tracer_tables = [Table.read(hdu) for hdu in hdu_list if hdu.header.get('SACCTYPE') == 'tracer']
+        window_tables = [Table.read(hdu) for hdu in hdu_list if hdu.header.get('SACCTYPE') == 'window']
+        data_tables = [Table.read(hdu) for hdu in hdu_list if hdu.header.get('SACCTYPE') == 'data']
+        cov = [hdu for hdu in hdu_list if hdu.header.get('SACCTYPE') == 'cov']
+
+        # Pull out the classes for these components.
+        tracers = BaseTracer.from_tables(tracer_tables)
+        windows = BaseWindow.from_tables(window_tables)
+
+        # The lookup table is used to convert from ID numbers to
+        # Window objects.
+        lookup = {'window': windows}
+
+        # Collect together all the data points from the different sections
+        data = []
+        for table in data_tables:
+            data += DataPoint.from_table(table, lookup)
+
+        # Finally, take all the pieces that we have collected
+        # and add them all into this data set.
+        S = cls()
+        for tracer in tracers.values():
+            S.add_tracer_object(tracer)
+
+        # Add the data points manually instead of using the API, since we
+        # have already constructed them.
+        for d in data:
+            S.data.append(d)
+
+        # Assume there is only a single covariance extension,
+        # if there are any
+        if cov:
+            S.add_covariance(BaseCovariance.from_hdu(cov[0]))
+
+        return S
+
+    #
+    # Methods below here are helper functions for specific types of data.
+    # We can add more of them as it becomes clear what people need.
+    #
+    #
+
+    def _get_2pt(self, data_type, bin1, bin2, return_cov, angle_name):
+        # Internal helper method for get_ell_cl and get_theta_xi
+        ind = self.indices(data_type, (bin1, bin2))
+
+        mu = np.array(self.mean[ind])
+        angle = np.array(self._get_tags_by_index([angle_name], ind)[0])
+
+        if return_cov:
+            if self.covariance is None:
+                raise ValueError("This sacc data does not have a covariance attached")
+            cov_block = self.covariance.get_block(ind)
+            return angle, mu, cov_block
+        else:
+            return angle, mu
+
+    def get_ell_cl(self, data_type, bin1, bin2, return_cov=False):
```

Comment:
Same comment as above for the case of swapping bin1 and bin2

Reply:
Maybe also worth noting that these functions cannot be used to retrieve covariance matrix elements between two different power spectra. This might limit their usefulness.

Reply:
Yeah, I was really only thinking that return_cov would be useful for plotting.

Reply:
sure, and that's fine